testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/saml/LogoutTest.java

```diff
@@ -195,6 +215,110 @@ public void testFrontchannelLogoutInSameBrowser() {
+            assertNotNull(backchannelLogoutReceiver.getLogoutRequest().getSignature());
+            // check nameID
+            assertEquals(logoutRequest.getNameID().getValue(), bburkeUser.getUsername());
+        } catch (Exception ex) {
+            fail("unexpected error");
+        }
+    }
+
+    /** Logout triggered with SOAP binding, request is properly signed */
+    @Test
+    public void testSoapBackchannelLogoutFromSamlClient() {
+        try {
+            adminClient.realm(REALM_NAME)
+                    .clients().get(salesSigRep.getId())
+                    .update(ClientBuilder.edit(salesSigRep)
+                            .frontchannelLogout(false)
+                            .attribute(SamlConfigAttributes.SAML_FORCE_NAME_ID_FORMAT_ATTRIBUTE, "true") // Force NameID to username
+                            .attribute(SamlConfigAttributes.SAML_NAME_ID_FORMAT_ATTRIBUTE, "username") // Force NameID to username
+                            .build());
+
+            SAMLDocumentHolder samlLogoutResponse = prepareLogIntoTwoAppsSig()
+                    .clearCookies() // remove cookies, since SOAP calls do not embed cookie normally
+                    .logoutRequest(getAuthServerSamlEndpoint(REALM_NAME), SAML_CLIENT_ID_SALES_POST_SIG, SOAP)
+                    .nameId(nameIdRef::get)
+                    .sessionIndex(sessionIndexRef::get)
+                    .signWith(SAML_CLIENT_SALES_POST_SIG_PRIVATE_KEY, SAML_CLIENT_SALES_POST_SIG_PUBLIC_KEY)
+                    .build()
+                    .getSamlResponse(SOAP);
+
+            assertThat(samlLogoutResponse.getSamlObject(), isSamlStatusResponse(JBossSAMLURIConstants.STATUS_SUCCESS));
+            assertSoapLogoutEvent(SAML_CLIENT_ID_SALES_POST_SIG);
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            fail("unexpected error");
+        }
+    }
+
+    /** Logout triggered with SOAP binding, request is wrongly not signed --> ensure an error is thrown */
+    @Test
+    public void testSoapBackchannelLogoutFromSamlClientUnsignedRequest() {
+        try {
+            adminClient.realm(REALM_NAME)
+                    .clients().get(salesSigRep.getId())
+                    .update(ClientBuilder.edit(salesSigRep)
+                            .frontchannelLogout(false)
+                            .attribute(SamlConfigAttributes.SAML_FORCE_NAME_ID_FORMAT_ATTRIBUTE, "true") // Force NameID to username
+                            .attribute(SamlConfigAttributes.SAML_NAME_ID_FORMAT_ATTRIBUTE, "username") // Force NameID to username
+                            .build());
+
+            try {
+                SAMLDocumentHolder samlLogoutResponse = prepareLogIntoTwoAppsSig()
+                        .clearCookies() // remove cookies, since SOAP calls do not embed cookie normally
+                        .logoutRequest(getAuthServerSamlEndpoint(REALM_NAME), SAML_CLIENT_ID_SALES_POST_SIG, SOAP, true)
```

Comment:
If I understand correctly, this test is testing this from spec:
```
Synchronous Bindings (Back-Channel)
Alternatively, the session participant MAY use a synchronous binding, such as the SOAP binding
[SAMLBind], to send the request directly to the identity provider. The identity provider SHOULD then
propagate any required logout messages to additional session participants as required using a
synchronous binding. The requester MUST authenticate itself to the identity provider, either by signing
the <LogoutRequest> or using any other binding-supported mechanism.
```
the last part `The requester MUST authenticate itself ....`.

This test is currently passing because of [this check](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/saml/SamlService.java#L301-L303). 

I am wondering, should this test pass also for `SAML_CLIENT_ID_SALES_POST` that does not have `"saml.client.signature"` set to `true`? It does not at the moment, should we add some check that requires a signature for `SOAP` binding regardless of the client settings?

Reply:
I'd say that the authentication of the requester is mandatory, but the means is not defined. It can be that logout requests are signed, but it could also be that there are some mutual TLS established on the link between the client and the IDP. Thus I would not make the message-level signature mandatory, but rather let the integrator decide how authentication is achieved through configuration of the client (`saml.client.signature`).

What do you think?

Reply:
ok, makes sense to me. Thanks for the explanation.

Reply:
I agree that authentication is out of the SAML SOAP Binding scope. I'm also comfortable with the current change in this aspect. Signatures and https should be enabled if a decent level of security is wanted (it's the same for all the SAML configuration).

But we have to keep in mind that clients can have weird things to allow the call from the server to execute the logout in then (for example basic auth, certificates,...). With the current implementation the server just sends the SOAP message with/without signature (nothing more) and this fact should be clear (maybe doc or something). I don't like either that SOAP is using default [SOAPConnectionFactory](https://github.com/keycloak/keycloak/blob/20.0.3/services/src/main/java/org/keycloak/protocol/saml/profile/util/Soap.java#L207-L209) to perform the SOAP call. It means using default `HttpURLConnection` instead of `SimpleHttp` (proxies, certs,... are not taken into account). But this is another issue, @mhajas WDYT? Do we file another issue for that?

The same for the server, it accepts logout requests with or without signature. Although prod configurations needs https and signatures for SAML to be minimally secure.

Reply:
@rmartinc this change sounds correct, however, I am not sure what changes we would require to make this happen. For sure, we can create an issue and then prioritize.

Reply:
@mhajas I created https://github.com/keycloak/keycloak/issues/17139 and assigned to me. I'll work on it when I have some time.

Reply:
Thank you @rmartinc